Modules/_ssl.c

```diff
@@ -6241,6 +6338,41 @@ _ssl_get_default_verify_paths_impl(PyObject *module)
     return NULL;
 }
 
+/*[clinic input]
+@critical_section
+_ssl.get_sigalgs
+[clinic start generated code]*/
+
+static PyObject *
+_ssl_get_sigalgs_impl(PyObject *module)
+/*[clinic end generated code: output=ab0791b63856854b input=bf74cdad3a19d29e]*/
+{
+#if OPENSSL_VERSION_NUMBER >= 0x30400000L
+    const char *sigalgs;
+    PyObject *sigalgs_str, *sigalgs_list;
+
+    if ((sigalgs = SSL_get1_builtin_sigalgs(NULL)) == NULL) {
+        PyErr_NoMemory();
+        return NULL;
+    }
+
+    if ((sigalgs_str = PyUnicode_DecodeASCII(sigalgs, strlen(sigalgs),
+                                             "strict")) == NULL) {
+        OPENSSL_free((void *)sigalgs);
+        return NULL;
+    }
```

Comment:
```suggestion
    sigalgs_str = PyUnicode_DecodeASCII(sigalgs, strlen(sigalgs), "strict");
    if (sigalgs_str == NULL) {
        OPENSSL_free((void *)sigalgs);
        return NULL;
    }
```

This should fit 80 chars. By the way, why can we use DecodeASCII but we use FSDefault when we get a single signature name? Can't we use both `PyUnicode_DecodeFSDefault`? (or both `PyUnicode_DecodeASCII`)

Reply:
I'd actually prefer to stick with PyUnicode_DecodeFSDefault here. Since the strings provided by OpenSSL should always be plain ASCII right now, it really won't change what gets returned. The DecodeASCII was something I think you suggested when reviewing `_ssl__SSLContext_get_groups_impl` because it provides a more explicit check if the string from OpenSSL contained non-ASCII, but I'm not sure how necessary that check is.

Following a pattern I saw elsewhere in the SSL module, my original code used DecodeFSDefault when returning the selected group in `_ssl__SSLSocket_group_impl`, and I stuck with that pattern in `ssl_socket_signame_impl`. So, it would be more consistent to use that here as well.

If I do make this change, should I change `_ssl__SSLContext_get_groups_impl` as well for consistency? I see that code is actually setting an SSL error as well, which may not be the best choice given any error is coming from Python in this case and should have already set the Python exception.

Reply:
> The DecodeASCII was something I think you suggested when reviewing _ssl__SSLContext_get_groups_impl because it provides a more explicit check if the string from OpenSSL contained non-ASCII

Oh? did I?

> So, it would be more consistent to use that here as well

Yes.

> If I do make this change, should I change _ssl__SSLContext_get_groups_impl as well for consistency

Yeah, let's change it, it doesn't really matter now as this is new code. 

> Following a pattern I saw elsewhere in the SSL module, my original code used DecodeFSDefault

Let's get back to DecodeFSDefault. Sorry for my previous review.

Reply:
Ok - I checked in the `get_sigalgs()` change to use DecodeFSDefault. Just to confirm, is it ok to make the get_groups() change in this current PR? I'll do it as a separate commit just to isolate it a bit more.

Reply:
Yes.

Reply:
Ok - done.